client/src/components/ProfileImageUploader.tsx

```diff
@@ -0,0 +1,39 @@
+import  { ChangeEvent } from 'react';
+import { ReactComponent as GalleryIcon } from '../assets/gallery_icon.svg';
+import placeholderPath from '../assets/pfp_placeholder.jpg';
+
+interface ProfileImageUploaderProps {
+  pfp: string;
+  setPfp: (pfp: string) => void;
+}
+
+const ProfileImageUploader = ({ pfp, setPfp }: ProfileImageUploaderProps) => {
+  const handleImageUpload = (event: ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setPfp(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+
+  return (
+    <label className="relative flex flex-col items-center self-center group">
+      <input
+        type="file"
+        className="absolute inset-0 w-full h-full opacity-0 cursor-pointer"
+        onChange={handleImageUpload}
+      />
+      <div className="w-32 h-32 rounded-full bg-gray-200 flex items-center justify-center overflow-hidden mt-8 relative border-2 border-primary-yellow">
+        <img src={pfp !== '' ? pfp : placeholderPath} alt="Profile" className="w-full h-full object-cover" />
```

Comment:
a co jeśli `pfp` to null? wtedy też będzie truthy

Reply:
pfp nie będzie null bo jest inicjalizowaine przez useState na ''